### DISCUSSION TITLE
Target URL 'subdomain.domain.com' has an insecure scheme 'http', only the 'https' and 'wss' schemes are supported so session cookies can be transmitted securely

### DISCUSSION THREAD

**Discussion by khadanja** [authelia/authelia#7420] [Troubleshooting] [answered]
My set up had to change as ISP now uses CGNAT, cloudflare tunnel client in docker on same host. Tunnel works for the protected app if I remove the protected app config from the advanced tab in npm so tunnel seems ok but when I add it back as per docs, I get above error in log & 403 Forbidden openresty in browser. I have already tried what I can but no luck. 
I can paste snippet files if requires but they are exactly like specified here -
https://github.com/authelia/authelia/discussions/4627#discussioncomment-4479337

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It seems like you are encountering a 403 Forbidden error when trying to access your protected application through a Cloudflare tunnel. This issue can be related to several factors, including misconfigurations in your NGINX Proxy Manager (NPM) advanced settings or issues with the Cloudflare tunnel setup.

Here are a few steps to troubleshoot and resolve the issue:

1. **Check Cloudflare Tunnel Configuration:**
   Ensure that your Cloudflare tunnel is correctly configured and that it routes traffic to the correct internal IP and port of your NGINX Proxy Manager.

2. **Verify NGINX Proxy Manager Advanced Settings:**
   Double-check the advanced settings in NPM for the protected application. Ensure that the configuration snippets are correctly included and that there are no typos or misconfigurations.

3. **Review Snippet Files:**
   Since you mentioned that your snippet files are exactly as specified in the documentation, ensure that they include the necessary headers and proxy settings. Here is an example of what the snippets should look like:

   ```nginx
   location / {
       include /snippets/proxy.conf;
       include /snippets/authelia-authrequest.conf;
       proxy_pass $forward_scheme://$server:$port;
   }
   ```

4. **Check for WebSocket Configuration:**
   If your application uses WebSockets, ensure that the necessary WebSocket headers are included in your configuration. You can add the following lines to your advanced settings:

   ```nginx
   proxy_set_header Upgrade $http_upgrade;
   proxy_set_header Connection "upgrade";
   ```

5. **Adjust Trusted Proxies:**
   Ensure that the IP ranges for trusted proxies are correctly set in your `proxy.conf` file. This is crucial if you are using Cloudflare, as it forwards the original client IP. Here is an example configuration:

   ```nginx
   set_real_ip_from 172.26.0.0/16; # Adjust this to match your network setup
   real_ip_header CF-Connecting-IP;
   real_ip_recursive on;
   ```

6. **Check Authelia Logs:**
   Review the Authelia logs for any errors or warnings that might indicate what is causing the 403 Forbidden error. This can provide more insight into what might be going wrong.

If you continue to face issues, please provide the specific error logs from Authelia and NGINX, as well as the exact configuration snippets you are using. This will help in diagnosing the problem more accurately.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
You will need to ensure either all connections from the tunnel use HTTPS when connecting to your proxy, or have cloudflare tunnel inject the X-Forwarded-Scheme header.
